sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/directconnectivity/GatewayServiceConfigurationReaderTest.java

```diff
@@ -118,37 +113,21 @@ public static void validateSuccess(Mono<DatabaseAccount> observable) {
     }
 
-    public static void validateSuccess(Mono<DatabaseAccount> observable, DatabaseAccount expectedDatabaseAccount) {
-        TestSubscriber<DatabaseAccount> testSubscriber = new TestSubscriber<DatabaseAccount>();
-
-        observable.subscribe(testSubscriber);
-        testSubscriber.awaitTerminalEvent(TIMEOUT, TimeUnit.MILLISECONDS);
-        testSubscriber.assertNoErrors();
-        testSubscriber.assertComplete();
-        testSubscriber.assertValueCount(1);
-        DatabaseAccount databaseAccount = testSubscriber.values().get(0);
-        assertThat(databaseAccount.getId()).isEqualTo(expectedDatabaseAccount.getId());
-        assertThat(BridgeInternal.getAddressesLink(databaseAccount))
-                .isEqualTo(BridgeInternal.getAddressesLink(expectedDatabaseAccount));
-        assertThat(databaseAccount.getWritableLocations().iterator().next().getEndpoint())
-                .isEqualTo(expectedDatabaseAccount.getWritableLocations().iterator().next().getEndpoint());
-        assertThat(BridgeInternal.getSystemReplicationPolicy(databaseAccount).getMaxReplicaSetSize())
-                .isEqualTo(BridgeInternal.getSystemReplicationPolicy(expectedDatabaseAccount).getMaxReplicaSetSize());
-        assertThat(BridgeInternal.getSystemReplicationPolicy(databaseAccount).getMaxReplicaSetSize())
-                .isEqualTo(BridgeInternal.getSystemReplicationPolicy(expectedDatabaseAccount).getMaxReplicaSetSize());
-        assertThat(BridgeInternal.getQueryEngineConfiuration(databaseAccount))
-                .isEqualTo(BridgeInternal.getQueryEngineConfiuration(expectedDatabaseAccount));
+    private void setBackgroundRefreshLocationTimeIntervalInMS(GlobalEndpointManager globalEndPointManager, int millSec) throws Exception {
+        Field backgroundRefreshLocationTimeIntervalInMSField = GlobalEndpointManager.class.getDeclaredField("backgroundRefreshLocationTimeIntervalInMS");
+        backgroundRefreshLocationTimeIntervalInMSField.setAccessible(true);
+        backgroundRefreshLocationTimeIntervalInMSField.setInt(globalEndPointManager, millSec);
     }
 
-    private HttpResponse getMockResponse(String databaseAccountJson) {
-        HttpResponse httpResponse = Mockito.mock(HttpResponse.class);
-        Mockito.doReturn(200).when(httpResponse).statusCode();
-        Mockito.doReturn(Flux.just(ByteBufUtil.writeUtf8(ByteBufAllocator.DEFAULT, databaseAccountJson)))
-                .when(httpResponse).body();
-        Mockito.doReturn(Mono.just(databaseAccountJson))
-                .when(httpResponse).bodyAsString();
+    private GatewayServiceConfigurationReader getServiceConfigurationReader(RxDocumentClientImpl rxDocumentClient) throws Exception {
+        Field serviceConfigurationReaderField = RxDocumentClientImpl.class.getDeclaredField("gatewayConfigurationReader");
+        serviceConfigurationReaderField.setAccessible(true);
+        return (GatewayServiceConfigurationReader) serviceConfigurationReaderField.get(rxDocumentClient);
```

Comment:
we have util helper methods for this:
please see this
`com.azure.cosmos.implementation.directconnectivity.get(.)`

Reply:
done